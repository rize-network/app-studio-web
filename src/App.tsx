import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
