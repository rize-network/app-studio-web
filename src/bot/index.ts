import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
