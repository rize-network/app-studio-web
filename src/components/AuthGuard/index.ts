export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
