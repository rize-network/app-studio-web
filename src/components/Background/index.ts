export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
