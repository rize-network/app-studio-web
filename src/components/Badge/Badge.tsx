import React from 'react';
import { BadgeProps } from './Badge/Badge.props';
import BadgeView from './Badge/Badge.view';

export const Badge = ({
  content,
  shape,
  colorScheme,
  position,
  variant,
  size,
  styles,
}: BadgeProps) => (
  <BadgeView
    content={content}
    shape={shape}
    colorScheme={colorScheme}
    position={position}
    variant={variant}
    size={size}
    styles={styles}
  />
);
