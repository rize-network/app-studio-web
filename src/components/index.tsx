export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
