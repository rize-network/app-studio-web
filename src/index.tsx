import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
