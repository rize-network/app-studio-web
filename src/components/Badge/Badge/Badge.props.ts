import { Size, Variant, Shape, Position, BadgeStyles } from './Badge.type';

export interface BadgeProps {
  content: string | number;
  variant?: Variant;
  colorScheme?: string;
  position?: Position;
  size?: Size;
  shape?: Shape;
  styles?: BadgeStyles;
  [x: string]: any;
}
