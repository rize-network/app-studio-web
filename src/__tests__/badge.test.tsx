import React from 'react';
import renderer from 'react-test-renderer';
import { Badge } from 'src/components';

import { cleanup, render, screen } from '@testing-library/react';
import { Position, Shape, Size } from 'src/components/Badge/Badge/Badge.type';

afterEach(() => {
  cleanup();
});

test('renders Badge component', () => {
  render(<Badge content="default" />);
  const BadgeElement = screen.getByRole('badge');
  expect(BadgeElement).toBeInTheDocument();
});

describe('Badge Shape Variants', () => {
  test.each([
    ['sharp', '0px'],
    ['rounded', '4px'],
    ['pillShaped', '24px'],
  ])('applies correct border-radius for shape %s', (shape, expectedRadius) => {
    render(<Badge content="test" shape={shape as Shape} />);
    const badgeElement = screen.getByRole('badge');
    expect(badgeElement).toHaveStyle({ borderRadius: expectedRadius });
  });
});

describe('Badge Positioning', () => {
  test.each([
    ['top-right', { top: '0', right: '0' }],
    ['top-left', { top: '0', left: '0' }],
    ['bottom-right', { bottom: '0', right: '0' }],
    ['bottom-left', { bottom: '0', left: '0' }],
  ])('positions badge correctly at %s', (position, expectedStyles) => {
    render(<Badge content="test" position={position as Position} />);
    const badgeElement = screen.getByRole('badge');
    expect(badgeElement).toHaveStyle(expectedStyles);
  });
});

describe('Badge Sizes', () => {
  test.each([
    ['xs', '20px'],
    ['sm', '24px'],
    ['md', '28px'],
    ['lg', '32px'],
    ['xl', '36px'],
  ])('applies correct size for %s size', (size, expectedMinWidth) => {
    render(<Badge content="test" size={size as Size} />);
    const badgeElement = screen.getByRole('badge');
    expect(badgeElement).toHaveStyle({ minWidth: expectedMinWidth });
  });
});

describe('Badge Custom Styles', () => {
  test('applies custom styles correctly', () => {
    const customStyles = {
      container: {
        backgroundColor: 'transparent',
        borderColor: 'purple',
      },
      text: {
        color: 'purple',
      },
    };

    render(<Badge content="test" views={customStyles} />);
    const badgeElement = screen.getByRole('badge');
    const badgeText = screen.getByRole('badgeText');

    expect(badgeElement).toHaveStyle({
      backgroundColor: 'transparent',
      borderColor: 'purple',
    });
    expect(badgeText).toHaveStyle({
      color: 'purple',
    });
  });
});

test('Badge to match snapshot', () => {
  const tree = renderer.create(<Badge content="default" />).toJSON();
  expect(tree).toMatchSnapshot();
});
