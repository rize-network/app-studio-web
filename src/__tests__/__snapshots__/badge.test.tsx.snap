// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge to match snapshot 1`] = `
<div
  className="sc-bdfCDU gXUExN"
  color="color.white"
  display="flex"
  height="28px"
  role="badge"
  width="fit-content"
>
  <div
    className="sc-bdfCDU FxXNY"
    fontSize={14}
    fontStyle="normal"
    fontWeight={400}
    role="badgeText"
    textDecoration="none"
  >
    default
  </div>
</div>
`;
